cmd/ctr/run_unix.go

```diff
@@ -54,66 +61,61 @@ func setHostNetworking() containerd.SpecOpts {
 
 func newContainer(ctx gocontext.Context, client *containerd.Client, context *cli.Context) (containerd.Container, error) {
 	var (
-		err             error
-		checkpointIndex digest.Digest
-
-		ref          = context.Args().First()
-		id           = context.Args().Get(1)
-		args         = context.Args()[2:]
-		tty          = context.Bool("tty")
-		labelStrings = context.StringSlice("label")
+		ref  = context.Args().First()
+		id   = context.Args().Get(1)
+		args = context.Args()[2:]
 	)
 
-	labels := labelArgs(labelStrings)
-
 	if raw := context.String("checkpoint"); raw != "" {
-		if checkpointIndex, err = digest.Parse(raw); err != nil {
+		checkpointIndex, err := digest.Parse(raw)
+		if err != nil {
 			return nil, err
 		}
-	}
-	image, err := client.GetImage(ctx, ref)
-	if err != nil {
-		return nil, err
+		if checkpointIndex != "" {
+			return client.NewContainer(ctx, id, containerd.WithCheckpoint(v1.Descriptor{
+				Digest: checkpointIndex,
+			}, id))
+		}
 	}
 
-	if checkpointIndex == "" {
-		opts := []containerd.SpecOpts{
-			containerd.WithImageConfig(ctx, image),
-			withEnv(context),
-			withMounts(context),
-		}
-		if len(args) > 0 {
-			opts = append(opts, containerd.WithProcessArgs(args...))
-		}
-		if tty {
-			opts = append(opts, withTTY())
-		}
-		if context.Bool("net-host") {
-			opts = append(opts, setHostNetworking())
-		}
-		spec, err := containerd.GenerateSpec(opts...)
+	var (
+		opts  []containerd.SpecOpts
+		cOpts []containerd.NewContainerOpts
+	)
+	cOpts = append(cOpts, containerd.WithContainerLabels(labelArgs(context.StringSlice("label"))))
+	if context.Bool("rootfs") {
```

Comment:
This logic is pretty odd. A rootfs flag that doesn't take an argument, then uses the image argument as the path? Let's find another way to do this.